server/src/main/java/org/elasticsearch/index/mapper/vectors/VectorEncoderDecoder.java

```diff
@@ -61,13 +70,20 @@ public static float getMagnitude(Version indexVersion, BytesRef vectorBR, float[
      * @param vectorBR - dense vector encoded in BytesRef
      * @param vector - array of floats where the decoded vector should be stored
      */
-    public static void decodeDenseVector(BytesRef vectorBR, float[] vector) {
+    public static void decodeDenseVector(Version indexVersion, BytesRef vectorBR, float[] vector) {
         if (vectorBR == null) {
             throw new IllegalArgumentException(DenseVectorScriptDocValues.MISSING_VECTOR_FIELD_MESSAGE);
         }
-        ByteBuffer byteBuffer = ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length);
-        for (int dim = 0; dim < vector.length; dim++) {
-            vector[dim] = byteBuffer.getFloat((dim * Float.BYTES) + vectorBR.offset);
+        if (indexVersion.onOrAfter(LITTLE_ENDIAN_FLOAT_STORED_INDEX_VERSION)) {
+            FloatBuffer fb = ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length)
+                .order(ByteOrder.LITTLE_ENDIAN)
+                .asFloatBuffer();
+            fb.get(vector);
+        } else {
+            ByteBuffer byteBuffer = ByteBuffer.wrap(vectorBR.bytes, vectorBR.offset, vectorBR.length);
+            for (int dim = 0; dim < vector.length; dim++) {
+                vector[dim] = byteBuffer.getFloat((dim * Float.BYTES) + vectorBR.offset);
+            }
```

Comment:
Could `.asFloatBuffer()` not be used for both little and big endian?

Reply:
`.asFloatBuffer()` is marginally slower for `BE`. These implementations are the fastest I could get them.